### PR TITLE
ft: add support for Cloud Events metadata headers

### PR DESCRIPTION
**Pull request by hauleth**

### CODE REVIEW

**Comment by hauleth** [comment]
@chasers ready to be merged

File: test/logflare_web/controllers/log_controller_test.exs [hunk 1]
```diff
@@ -65,6 +65,24 @@ defmodule LogflareWeb.LogControllerTest do
...
+
+      conn =
+        conn
+        |> put_req_header("x-api-key", user.api_key)
+        |> put_req_header("ce-foo", "bar")
+        |> post(Routes.log_path(conn, :cloud_event, source: source.token), @valid)
+
+      assert json_response(conn, 200) == %{"message" => "Logged!"}
+      assert_receive {:logs, [log]}, 3000
+      assert %{"metadata" => %{"cloud_event" => ce}} = log.params
```
Comment (Ziinc) [inline test/logflare_web/controllers/log_controller_test.exs:82]: don't need to nest under `metadata`

File: lib/logflare_web/controllers/log_controller.ex [hunk 2]
```diff
@@ -190,4 +209,11 @@ defmodule LogflareWeb.LogController do
     |> put_view(LogflareWeb.LogView)
     |> render("index.json", message: errors)
   end
+
+  defp extract_cloud_events(%Plug.Conn{} = conn) do
+    # XXX: What should happen in case of duplicated header?
+    for {"ce-" <> header, data} <- conn.req_headers, into: %{} do
```
Comment (Ziinc) [inline lib/logflare_web/controllers/log_controller.ex:215]: we also need to replace all hyphens with an underscore

File: test/logflare_web/controllers/log_controller_test.exs (hunk 1 above)
Comment (Ziinc) [reply test/logflare_web/controllers/log_controller_test.exs:82]: placing the `cloud_event` key on top level would be good.

**Review by Ziinc — DISMISSED** [review]

**Review by Ziinc — APPROVED** [review]

**Pull request merged**